include/common/core/logging.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
```

Comment:
Why this approach over BOOST? Because it's simple.

Also, I didn't see how to conditionally compile BOOST logging.

Reply:
you'd kind of do both, i think. in the `#ifdef ENABLE_LOGGING` section your log macro would forward to boost, otherwise blackhole it like you do today.

Reply:
I don't see the big benefit of boost yet.